src/common/telemetry-data-factory.ts

```diff
@@ -523,4 +538,43 @@ export class TelemetryDataFactory {
             enabled,
         };
     }
+
+    public forCompletedTestStep(
+        assessment: Assessment,
+        storeData: AssessmentStoreData,
+        step: Requirement,
+    ): RequirementStatusTelemetryData {
+        const assessmentData = assessment
+            .getVisualizationConfiguration()
+            .getAssessmentData(storeData);
+        const numInstances = this.getNumInstances(step, assessmentData);
+        const newStatus = storeData.assessments[assessment.key].testStepStatus;
+        return this.forRequirementStatus(
+            assessment.visualizationType,
+            step.key,
+            newStatus[step.key].stepFinalResult === ManualTestStatus.PASS,
+            numInstances,
+            step.getCompletedStepDetailsForTelemetry
+                ? step.getCompletedStepDetailsForTelemetry(assessment, storeData)
+                : undefined,
+        );
+    }
+
+    private getNumInstances(step: Requirement, assessmentData: AssessmentData): number {
```

Comment:
I think this should remain in completed-test-step-telemetry as this is not really relevant for telemetry-data-factory.

Reply:
> I think this should remain in completed-test-step-telemetry as this is not really relevant for telemetry-data-factory.

Agreed. Making it public in TelemetryDataFactory is what led to the null:check problems. The new iteration limits TelemetryDataFactory changes to adding a new optional parameter to `forRequirementStatus`